Diagrams

ianbjacobs opened the pull request:
- Added diagrams (based on conversations with @mattsaxon
- Part of the diagram discussion had been around the modeling, so cleaned up some of the modeling language and added points that @mattsaxon had wanted clarified.
- Added a requestPermission() method and marked it with issue 94 (instead of only having requestPermission() appear in an example)

ianbjacobs committed:
* Small fixes
* Also clarified that paymentResponse is not necessarily for successful
completion of "payment" in the sense of exchange of funds. For example,
it might involve merely returning card credentials. So reframed in terms
of successful user interaction.

ianbjacobs committed:
tweaks based on marcos suggestion (remove "user agent" from
"user agent implementers")

ianbjacobs committed:
Merge remote-tracking branch 'w3c/gh-pages' into gh-pages

ianbjacobs committed:
Merge remote-tracking branch 'w3c/gh-pages' into gh-pages

ianbjacobs committed:
respec fixes (except 5)

ianbjacobs committed:
More respec fixes, including better use of respec attributes,
addition of more dependencies

ianbjacobs committed:
More tweaks based on Marcos comments

ianbjacobs committed:
Edits from Marcos. Not all handled yet

ianbjacobs committed:
Additional cleanup

ianbjacobs committed:
more fixes!

ianbjacobs committed:
caps fix

ianbjacobs committed:
changes with AdamR to add clear()

ianbjacobs committed:
removed example in favor of clear()

ianbjacobs committed:
key order : original insertion order

ianbjacobs committed:
Add ref to issue 128 re: examples

ianbjacobs committed:
Merge remote-tracking branch 'w3c/gh-pages' into gh-pages

ianbjacobs committed:
webidl fix

ianbjacobs committed:
Accommodating images with some text changes and preparing for images

ianbjacobs committed:
more edits...prep for images

ianbjacobs committed:
Merge remote-tracking branch 'w3c/gh-pages' into diagrams

ianbjacobs committed:
Added images

ianbjacobs committed:
Added

adrianhopebailie commented on index.html:
```diff
@@ -242,28 +242,37 @@
+
+    <p class="issue" title="Issue 127 - Payment Manager / Handler"><a href=
+      "https://github.com/w3c/webpayments-payment-apps-api/issues/127">
+      Issue 97</a>. The distinction between Payment Handler and Payment Manager is not clear. There is a proposal to remove the phrase Payment Manager from the spec, renaming the "PaymentManager" interface the "PaymentHandler" interface.</p>
```
Suggest this issue can be closed so this could be removed. Happy to leave in until the issue is closed so shouldn't hold up this PR.

adrianhopebailie commented on index.html:
```diff
@@ -499,12 +542,13 @@
-          payment methods enabled by this Instrument.
+          member is a list of one or more
+	  <a>payment method identifiers</a> of the
+          <a>payment method</a>s enabled by this Instrument.
```
enabled -> supported

ianbjacobs committed:
Merge remote-tracking branch 'w3c/gh-pages' into diagrams

Resolved conflicts

ianbjacobs committed:
s/enabled/supported per AHB
https://github.com/w3c/webpayments-payment-apps-api/pull/132#discussion_r111591136

marcoscaceres commented:
Why did this PR include all the old commits from the other PR? Please make sure nothing got trashed. Always make sure your commit history only includes the actual changes as stated in the bug or this is going to end in 😭

ianbjacobs commented:
I am confident nothing was trashed. To compress the commit history do I need to rebase? 
When I don't, the commit history seems to grow and grow.

Ian

marcoscaceres commented:
@ianbjacobs, so, what's happening is that you need to push with force - also try avoid "merge" entirely. 

Your workflow should be:

```Bash
git checkout gh-pages
git pull gh-pages
git checkout -b "feature_or_bug_to_fix"
git commit -m "action(optional context): thing I did (closes #bug_number)" index.html 
``` 

If you see other PRs coming in after you sent your pull request:

```Bash 
git checkout gh-pages
git pull gh-pages 
git checkout "feature_or_bug_to_fix"
git rebase gh-pages
git push -f 
```

Last push there will prevent your history from growing. Then we should always squash merge here on Github - making sure the history massage makes sense.

Again, happy to help with this stuff (git can be annoying and confusing) - but please let's make a real effort to always check things are in a good state in the repo before merging.

marcoscaceres commented:
I'd also recommend reading up on git rebase: https://help.github.com/articles/about-git-rebase/

There are some great short videos on Youtube also - the can be really helpful in cleaning up commit history.

adrianhopebailie commented:
@marcoscaceres, @ianbjacobs is also making multiple commits as part of a PR, often in response to reviews.

I think we need to agree on a process for merging the PR if it legitimately contains multiple commits (i.e. not commits that also appear in other PRs).

I suggest we just use GitHub's Squash and merge

ianbjacobs commented:
@marcoscaceres,

Thank you for the workflow details. I was doing all that except:

 1) The structure of the commit message
 2) The rebase.

I will start doing that. Thanks!

Ian
